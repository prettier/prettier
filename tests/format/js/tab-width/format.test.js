runFormatTest(import.meta, ["babel", "flow", "typescript"]);
runFormatTest(import.meta, ["babel", "flow", "typescript"], { tabWidth: 4 });
