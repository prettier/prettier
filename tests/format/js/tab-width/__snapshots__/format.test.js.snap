// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class.js - {"tabWidth":4} format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
tabWidth: 4
                                                                                | printWidth
=====================================input======================================
class A {
  method() {
    var x = 1;
    while(typeof x == "number" || typeof x == "string") {
        x = x + 1;
        if (true) x = "";
    }
    var z = x;
  }
}

=====================================output=====================================
class A {
    method() {
        var x = 1;
        while (typeof x == "number" || typeof x == "string") {
            x = x + 1;
            if (true) x = "";
        }
        var z = x;
    }
}

================================================================================
`;

exports[`class.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
class A {
  method() {
    var x = 1;
    while(typeof x == "number" || typeof x == "string") {
        x = x + 1;
        if (true) x = "";
    }
    var z = x;
  }
}

=====================================output=====================================
class A {
  method() {
    var x = 1;
    while (typeof x == "number" || typeof x == "string") {
      x = x + 1;
      if (true) x = "";
    }
    var z = x;
  }
}

================================================================================
`;

exports[`nested-functions.spec.js - {"tabWidth":4} format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
tabWidth: 4
                                                                                | printWidth
=====================================input======================================
const c = () => {};

function a() {
  return function b() {
    return () => {
      return function() {
        return c;
      }
    }
  }
}

=====================================output=====================================
const c = () => {};

function a() {
    return function b() {
        return () => {
            return function () {
                return c;
            };
        };
    };
}

================================================================================
`;

exports[`nested-functions.spec.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
const c = () => {};

function a() {
  return function b() {
    return () => {
      return function() {
        return c;
      }
    }
  }
}

=====================================output=====================================
const c = () => {};

function a() {
  return function b() {
    return () => {
      return function () {
        return c;
      };
    };
  };
}

================================================================================
`;
